docs/lab/08/index.md

```diff
@@ -0,0 +1,435 @@
+            warn!("write error: {:?}", e);
+            break;
+        }
+    };
+}
+```
+
+#### Starting an UDP server
+
+Similarly to the TCP server, we create a new socket, this time one that handles UDP connections.
+
+```rust
+let mut rx_buffer = [0; 4096];
+let mut rx_metadata_buffer = [PacketMetadata::EMPTY; 3];
+let mut tx_buffer = [0; 4096];
+let mut tx_metadata_buffer = [PacketMetadata::EMPTY; 3];
+
+let mut buf = [0u8; 4096];
+
+let mut socket = UdpSocket::new(
+    stack, &mut rx_metadata_buffer, &mut rx_buffer, &mut tx_metadata_buffer, &mut tx_buffer
+);
+```
+
+Next, we bind our socket to a port (for example port `1234`). Note that the `bind` function is not a future, since we don't need to wait for connections before initiating a transmission.
+
+```rust
+if let Err(e) = socket.bind(1234) {
+    warn!("accept error: {:?}", e);
+    continue;
+}
+```
+
+Now, we can read and write to the socket, and any endpoint that connects to the port can send/receive data.
+
+```rust
+// example read
+loop {
+    match socket.recv_from(&mut buf).await {
+        Ok((n, endpoint)) => {
+            info!("Received from {:?}: {:?}", endpoint, from_utf8(&buf[..n]).unwrap());
+        },
+        Err(_) => {
+            info!("An error occurred when receiving the packet!");
+        }
+    }
+}
+
+// example write
+let buffer = "hello\n".as_bytes();
+match socket.send_to(&buffer, IpEndpoint::new(IpAddress::v4(192, 168, 100, 45), 1234)).await {
+    Ok(()) => {
+        info!("sent")
+    }
+    Err(e) => {
+        warn!("send error: {:?}", e);
+    }
+}
+```
+
+## Exercises
+
+1. The lab skeleton contains an example of setting up the Wi-Fi and connecting to an access point.
+- Scan for local networks. (**1p**)
+- Modify the code in order to connect to an existing network. Once connected, determine the IP address that has been assigned to the Pico through DHCP and the default gateway of the network. (**1p**)
+:::tip
+You can use `stack.config_v4()` to find the assigned IP address and default gateway. 
+:::
+- Use a static IP address instead of a dynamic one. You can use the IP determined at the previous point. (**1p**)
+
+2. Test the example by connecting your computer to the Pico through Wi-Fi. Use `netcat` or any alternative to send a TCP packet to the Pico server. You should receive back the same string. (**1p**)
+
+:::tip
+On Windows, you can use [NCat](https://nmap.org/download.html#windows). The command for establishing a TCP connection is:
+```shell
+ncat [ip_address] [port]
+```
+To send a packet, write any string in the terminal and press enter. You should receive back the same string, since this is what our server is currently configured to do.
+You can run `ncat -h` to see a list of other available parameters for this command.
+:::
+
+3. Modify the way the server on the Pico deals with connections. Send a message through Wi-Fi to your computer whenever a button is pressed. It's better to use UDP for this. (Why?) (**2p**)
```

Comment:
The Why is a question for us the reviewers? If so:
- the packet os received fully, not fragmented; tcp might fragment the packet
- the local network should not loose a lot of packets

Reply:
I was debating whether to have the students try and answer the question or not. I removed it in the end.